src/codeview.js

```diff
@@ -453,29 +453,32 @@ var Codeview = GObject.registerClass({
 
         if (this._numErrors !== results.length) {
             this._numErrors = results.length;
-            if (this._numErrors)
-                SoundServer.getDefault().play('codeview/code-error-appearance');
-            else
-                SoundServer.getDefault().play('codeview/code-error-disappearance');
+            // Only play sounds when user compiles the code
+            if (userInitiated) {
+                if (this._numErrors)
+                    SoundServer.getDefault().play('codeview/code-error-appearance');
+                else
+                    SoundServer.getDefault().play('codeview/code-error-disappearance');
+            }
         }
 
         this._recordErrorMetrics(results)
```

Comment:
I think we should probably only record metrics if the error is user initiated, as well.